pkg/vcdclient/auth.go

```diff
@@ -86,12 +115,77 @@ func (config *VCDAuthConfig) GetPlainClientFromSecrets() (*govcd.VCDClient, erro
 	return vcdClient, nil
 }
 
-func NewVCDAuthConfigFromSecrets(host string, user string, secret string, org string, insecure bool) *VCDAuthConfig {
+type tokenResponse struct {
+	AccessToken  string `json:"access_token"`
+	TokenType    string `json:"token_type"`
+	ExpiresIn    int32  `json:"expires_in"`
+	RefreshToken string `json:"refresh_token"`
+}
+
+func (config *VCDAuthConfig) getAccessTokenFromRefreshToken(isSysadminUser bool) (*tokenResponse, *http.Response, error) {
+	accessTokenUrl := fmt.Sprintf("%s/oauth/provider/token", config.Host)
+	if !isSysadminUser {
+		accessTokenUrl = fmt.Sprintf("%s/oauth/tenant/%s/token", config.Host, config.Org)
+	}
+
+	payload := url.Values{}
+	payload.Set("grant_type", "refresh_token")
+	payload.Set("refresh_token", config.RefreshToken)
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+	r, err := http.NewRequest("POST", accessTokenUrl, strings.NewReader(payload.Encode())) // URL-encoded payload
```

Comment:
Since we have variables `r` and `res`, could we rename them to something more descriptive like `postResponse` and `clientResponse`?